src/bot.ts

```diff
@@ -4,60 +4,232 @@ import { Bot, Context, InputFile, InlineKeyboard } from "grammy";
 const { BOT_TOKEN: token = "" } = process.env;
 export const bot = new Bot(token);
 
-//сховище для стоврення записки
+let isGetFile = false;
+let isDeleteNote = false;
+let isEditNote = false;
+
+//сховище для стоврення та зміни записки
 const memoData = {};
 
+// id для зміни файлу
+let editFileId = 0;
+
+const instance = axios.create({
+    baseURL: "https://sr-kpi-api-development.up.railway.app",
+});
+
+function decodeFilename(contentDisposition) {
+    const match = contentDisposition.match(/filename\*?=['"]?([^;]*)['"]?/);
+    if (match && match[1]) {
+        return decodeURIComponent(match[1].replace(/UTF-8''/, ""));
+    }
+    return "unknown_filename";
+}
+
 // команда для Вітаннячка
 bot.command("start", async (ctx) => {
+    isGetFile = false;
+    isDeleteNote = false;
+    isEditNote = false;
+    delete memoData[ctx.chat.id];
+
     await ctx.reply("Вітаю! \nЦей Бот допомагає створити службову записку. Для створення натисніть /createnote ");
 });
 
 bot.command("createnote", async (ctx) => {
+    isGetFile = false;
+    isDeleteNote = false;
+    isEditNote = false;
+    delete memoData[ctx.chat.id];
+
     const keyboard = new InlineKeyboard().text("Службова записка", "service_memo").text("Подання", "submission").text("Звернення", "appeal");
 
     await ctx.reply("Будь ласка, виберіть тип документу:", { reply_markup: keyboard });
     memoData[ctx.chat.id] = { step: "choose_type" };
 });
 
-bot.callbackQuery("service_note", async (ctx) => {
-    memoData[ctx.chat.id].type = "Службова записка";
+bot.command("notelist", async (ctx) => {
+    isGetFile = false;
+    isDeleteNote = false;
+    isEditNote = false;
+    delete memoData[ctx.chat.id];
+
+    try {
+        const response = await instance.get("/documents/notes");
+
+        if (Array.isArray(response.data) && response.data.length === 0) {
+            await ctx.reply("Записів нема");
+        } else {
+            const formattedData = response.data
+                .map((item, index) => `${index + 1}\\) ${item.name.replaceAll(".", "\\.").replaceAll("_", "\\_")} \\- \`${item.id}\``)
+                .join("\n");
```

Comment:
extract data formatting to the separated function, your logic is duplicating

Reply:
Fixed